### PR TITLE
Balder/bool for summary too

### PR DESCRIPTION
**Pull request by baldersheim**
@bratseth and @geirst PR

### CODE REVIEW

**Review by bratseth — APPROVED** [review]
LGTM! Especially all the red ...

**Review by geirst — COMMENTED** [review]
👍 Have a nitpick and a comment, please have a look

File: searchsummary/src/vespa/searchsummary/docsummary/docsumfieldwriter.h [hunk 1]
```diff
@@ -46,20 +43,27 @@ class IDocsumFieldWriter
     static const vespalib::string _empty;
 };
 
+class IDocsumFW : public IDocsumFieldWriter
```
Comment (geirst) [inline searchsummary/src/vespa/searchsummary/docsummary/docsumfieldwriter.h]: It seems like this class is for those that do not care about GeneralResult* from the insertField() function. Consider giving it at more descriptive name and add a class comment. My suggestion is SimpleDocsumFieldWriter.

File: document/src/vespa/document/fieldvalue/boolfieldvalue.h [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
+// Copyright 2019 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
 
 #pragma once
 
 #include "fieldvalue.h"
 
 namespace document {
 
+/**
+ * Represent the value in a filed of type 'bool' which can be either true or false.
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply document/src/vespa/document/fieldvalue/boolfieldvalue.h]: Fixed in #8134

File: searchsummary/src/vespa/searchsummary/docsummary/docsumfieldwriter.h (hunk 1 above)
Comment (baldersheim) [reply searchsummary/src/vespa/searchsummary/docsummary/docsumfieldwriter.h]: Agreed, ISimpleDFW it is.

**Pull request merged**